Resources/Storage/GFAL2_StorageBase.py

```diff
@@ -396,31 +410,35 @@ def _getSingleFile( self, src_url, dest_file, disableChecksum = False ):
     :returns: S_ERROR( errStr ) in case of an error
               S_OK( size of file ) if copying is successful
     """
-    self.log.info( "GFAL2_StorageBase._getSingleFile: Trying to download %s to %s" % ( src_url, dest_file ) )
+
+    log = self.log.getSubLogger( "GFAL2_StorageBase._getSingleFile" )
+
+    log.info( "Trying to download %s to %s" % ( src_url, dest_file ) )
+    if disableChecksum:
+      log.warn( "checksum calculation disabled for transfers!" )
 
     destDir = os.path.dirname( dest_file )
 
     if not os.path.exists( destDir ):
-      self.log.debug( "GFAL2_StorageBase._getSingleFile: Local directory does not yet exist. Creating it", destDir )
+      log.debug( "Local directory does not yet exist. Creating it", destDir )
       try:
         os.makedirs( destDir )
-      except OSError, error:
-        errStr = "GFAL2_StorageBase._getSingleFile: Error while creating the destination folder"
-        self.log.exception( errStr, error )
-        return S_ERROR( errStr )
+      except OSError as error:
+        errStr = "Error while creating the destination folder"
+        log.exception( errStr, lException = error )
+        return S_ERROR( "%s: %s" % ( errStr, repr( error ) ) )
 
     res = self.__getSingleFileSize( src_url )
     if not res['OK']:
-      errStr = "GFAL2_StorageBase._getSingleFile: Error while determining file size: %s" % res['Message']
-      self.log.error( res['Message'] )
+      log.error( "Error while determining file size", res['Message'] )
       return res
 
     remoteSize = res['Value']
 
     # Set gfal2 copy parameters
     # folder is created and file exists, setting known copy parameters
     params = self.gfal2.transfer_parameters()
-    params.timeout = int( remoteSize / self.MIN_BANDWIDTH + 300 )
+    params.timeout = 10 + int( remoteSize / self.MIN_BANDWIDTH * 4 + 300 )
```

Comment:
In practice, we see that when there is a timeout, it really means sluggish link, so it should not be too bad :-) But I just sent a mail to Alejandro to see if these estimations are still good or need to be reviewed with gfal2